src/features/Applet/Schedule/Calendar/YearView/MonthCalendar/MonthCalendar.utils.ts

```diff
@@ -0,0 +1,34 @@
+import { DateLocalizer } from 'react-big-calendar';
+import { firstVisibleDay, lastVisibleDay } from 'react-big-calendar/lib/utils/dates';
+import { add } from 'date-fns';
+
+import { EN_WEEK_DAYS, FR_WEEK_DAYS, MILLISECONDS_PER_WEEK } from './MonthCalendar.const';
+import { MonthObject, Week } from './MonthCalendar.types';
+
+export const getMonthName = (date: Date, langLocale: Intl.LocalesArgument) =>
+  date.toLocaleString(langLocale, { month: 'long' });
+
+export const shortWeekDaysArray = (langLocale: Intl.LocalesArgument) =>
+  langLocale === 'en' ? EN_WEEK_DAYS : FR_WEEK_DAYS;
+
+export const createCalendar = (currentDate: string | Date, localizer: DateLocalizer) => {
+  const date = new Date(currentDate);
+  const first: Date = firstVisibleDay(date, localizer);
+  const last: Date = lastVisibleDay(date, localizer);
+  const weeksCount = Math.round(Math.abs(last.getTime() - first.getTime()) / MILLISECONDS_PER_WEEK);
+  const calendar: MonthObject = Object.assign({}, { date, first, last, weeks: [] });
```

Comment:
I believe more simple like this {...data, first, last}

Reply:
done